### PR TITLE
Sleep Feature 

### PR DESCRIPTION
**Pull request by ttarlov**
# PR Description
Please include a summary of the change:

:construction: HTML modifications/updates.

    - Adds divs for Water and Sleep data. 

:art: CSS modifications/updates.

    - Adds styling for the added divs as well as animations on load and on click for the tiles.

:crystal_ball: JS modifications/updates.

    - Adds Sleep Class to handle sleep data.
    - Adds Tests for Sleep Class. 
    - Re-organizes dom manipulation functions into seperate functions for displaying sleep, hydration and user data. 
    - Adds jiggle function. 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -28,17 +28,29 @@ <h3>Friends:</h3>
...
     </div>
     <div class="tile div4">
+        <h3 class="heading">Step Data</h3>
       <h3>Daily Step Goal: <span class="user-daily-step-goal-js"></span></h3>
-    </div>
-    <div class="tile div5">
       <h3>Other People's Average Step Goal:</h3>
       <h3><span class="step-goal-comparison-js"></span></h3>
     </div>
+    <div class="tile div5">
```
Comment (Karlfunhouse) [inline src/index.html]: Populating the Sleep Data Tile by interpolating all the info through our scripts.js file.

File: src/scripts.js [hunk 2]
```diff
@@ -11,22 +12,65 @@ let userDailyStepGoal = document.querySelector('.user-daily-step-goal-js');
 let userFriends = document.querySelector('.user-friends-js');
 let stepGoalComparison = document.querySelector('.step-goal-comparison-js');
 let waterConsumedByDay = document.querySelector('.water-consumed-by-day-js');
-let waterConsumedByWeek = document.querySelector('.water-consumed-by-week-js')
+let waterConsumedByWeek = document.querySelector('.water-consumed-by-week-js');
```
Comment (Karlfunhouse) [inline src/scripts.js]: adding querySelectors to populate the different tiles on the Dashboard based on where we want the data to appear.

File: src/scripts.js [hunk 3]
```diff
@@ -11,22 +12,65 @@ let userDailyStepGoal = document.querySelector('.user-daily-step-goal-js');
...
+let sleepQuality = document.querySelector('.sleep-quality-js');
+let lastWeekSleep = document.querySelector('.last-week-sleep-js');
+let recomendedSleep = document.querySelector('.sleep-percentage-js')
+let avgSleepQuality = document.querySelector('.avg-sleep-quality-js')
+let avgHoursSleptAllTime = document.querySelector('.all-time-avg-hours-slept-js')
+let parentDiv = document.querySelector('.parent')
+
+parent.addEventListener('click', makeItShake);
+
+function makeItShake(event) {
```
Comment (Karlfunhouse) [inline src/scripts.js]: nice load animation

File: src/scripts.js [hunk 4]
```diff
@@ -11,22 +12,65 @@ let userDailyStepGoal = document.querySelector('.user-daily-step-goal-js');
...
+  if (event.target.classList.contains('tile')) {
+    event.target.closest('.tile').classList.toggle('vibe')
+  }
+}
 
 displayUserInfo(user);
+displayWaterConsumtion();
```
Comment (Karlfunhouse) [inline src/scripts.js]: breaking down the massive function into smaller bites based on the different classes or data sets.  Nice

File: src/scripts.js [hunk 5]
```diff
@@ -11,22 +12,65 @@ let userDailyStepGoal = document.querySelector('.user-daily-step-goal-js');
...
+
+
+function displaySleepData() {
+  avgHoursSleptAllTime.innerHTML = `${sleep.calculateTotalAvgHoursSlept()} hrs`
+  avgSleepQuality.innerHTML = `${(((sleep.calculateAvgSleepQuality())/5)*100).toFixed(2)}%`
+  recomendedSleep.innerHTML = `${sleep.calculatePercentageOfRecommendedSleep("2019/09/22")}`
+  sleepQuality.innerHTML = `${(((sleep.calculateQualityOfSleepByDay("2019/09/22"))/5)*100).toFixed(2)} %`;
+  hoursSlept.innerHTML = `${sleep.calculateHoursSleptByDay("2019/09/22")}`;
+  lastWeekSleep.innerHTML =
+  `<p><u><span>Date </span><span> : Hours Slept</span> : <span>Sleep Quality</span></u></p>
```
Comment (Karlfunhouse) [inline src/scripts.js]: I know this is beefy and we can possibly refactor it in the future but it works for now so once we have our MVP achieved we can look back on these.

File: src/sleep.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
+class Sleep {
+  constructor(sleepData, id) {
+    this.data = sleepData;
+    this.id = id;
+    this.user = this.findUserSleepData();
+  };
+
+  findUserSleepData() {
```
Comment (Karlfunhouse) [inline src/sleep.js]: finding the user

File: src/sleep.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    return this.data.filter(user => user.userID === this.id);
+  }
+
+  calculateAvgSleepQuality() {
```
Comment (Karlfunhouse) [inline src/sleep.js]: calculates a user's average sleep quality

File: src/scripts.js (hunk 5 above)
Comment (ttarlov) [reply src/scripts.js]: We can absolutely refactor this in the future. I have a pretty good idea on how we could do that with a nice forEach:-)

**Pull request merged**